### PR TITLE
PHP8 - Update

### PR DESCRIPTION
**Pull request by iomidev**
PHP8

### CODE REVIEW

File: Services/Language/classes/class.ilLanguageFile.php [hunk 1]
```diff
@@ -34,87 +34,26 @@
 */
 class ilLanguageFile
 {
-    /**
-     * Created global file objects
-     * @static		array
-     */
     private static $global_file_objects = array();
```
Comment (sailorsilv) [inline Services/Language/classes/class.ilLanguageFile.php]: why not private static array.. here?

**Review by sailorsilv — APPROVED** [review]

File: Services/Language/classes/Setup/class.ilLanguageMetricsCollectedObjective.php [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
 /* Copyright (c) 2020 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
 
 use ILIAS\Setup;
 
 class ilLanguageMetricsCollectedObjective extends Setup\Metrics\CollectedObjective
 {
-    /**
-     * @var \ilSetupLanguage
-     */
     protected $il_setup_language;
```
Comment (katringross) [inline Services/Language/classes/Setup/class.ilLanguageMetricsCollectedObjective.php]: Im Setup ist der Typ der Setup Sprache so definiert: 
protected \ilSetupLanguage $lng;

siehe \src\Setup\ImplementationOfAgentFinder.php

File: Services/Language/classes/Setup/class.ilLanguagesInstalledAndUpdatedObjective.php [hunk 3]
```diff
@@ -1,14 +1,11 @@
...
 /* Copyright (c) 2019 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
 
 use ILIAS\Setup;
 
 class ilLanguagesInstalledAndUpdatedObjective extends ilLanguageObjective
 {
-    /**
-     * @var \ilSetupLanguage
-     */
     protected $il_setup_language;
```
Comment (katringross) [inline Services/Language/classes/Setup/class.ilLanguagesInstalledAndUpdatedObjective.php]: see last comment https://github.com/kroepelin-projekte/ILIAS/pull/1/files#r746264452

File: Services/Language/classes/Setup/class.ilLanguagesUpdatedObjective.php [hunk 4]
```diff
@@ -1,14 +1,11 @@
...
 /* Copyright (c) 2019 Richard Klees <richard.klees@concepts-and-training.de> Extended GPL, see docs/LICENSE */
 
 use ILIAS\Setup;
 
 class ilLanguagesUpdatedObjective implements Setup\Objective
 {
-	/**
-	 * @var \ilSetupLanguage
-	 */
 	protected $il_setup_language;
```
Comment (katringross) [inline Services/Language/classes/Setup/class.ilLanguagesUpdatedObjective.php]: same as https://github.com/kroepelin-projekte/ILIAS/pull/1/files#r746264452

File: Services/Language/classes/Setup/class.ilSetupLanguage.php [hunk 5]
```diff
@@ -375,34 +321,31 @@ protected function checkLanguage($a_lang_key, $scope = '')
...
      *
-     * @param    string      $content    expect an ILIAS lang-file
-     * @return   string      $content    content without header info OR false if no valid header was found
-     * @access   private
+     * $content    expect an ILIAS lang-file
+     *
      */
-    protected function cut_header($content)
+    protected function cut_header(array $content): array|bool
     {
```
Comment (katringross) [inline Services/Language/classes/Setup/class.ilSetupLanguage.php:328]: Here I am not sure, shouldn`t it be: protected function cut_header(string $content): string|bool ?

Before $content is named as string, but why use array_slice for an array? Doesn't make sense. 

So at the end I would go with your decision.

File: Services/Language/classes/Setup/class.ilSetupLanguage.php [hunk 6]
```diff
@@ -411,9 +354,9 @@ protected function flushLanguage($a_lang_key, $a_mode = 'all')
     /**
     * Delete languge data
     *
-    * @param	string		lang key
+    * $a_lang_key		lang key
     */
-    public static function _deleteLangData($a_lang_key, $a_keep_local_change)
+    public static function _deleteLangData(string $a_lang_key, $a_keep_local_change): void
     {
```
Comment (katringross) [inline Services/Language/classes/Setup/class.ilSetupLanguage.php:360]: bool $a_keep_local_change ?

File: Services/Language/classes/class.ilCachedLanguage.php [hunk 7]
```diff
@@ -10,34 +11,17 @@
 class ilCachedLanguage
 {
     protected $global_cache;
-    /**
```
Comment (katringross) [inline Services/Language/classes/class.ilCachedLanguage.php:13]: Type \ilGlobalCacheService 

see Services\GlobalCache\classes\class.ilGlobalCache.php

File: Services/Language/classes/class.ilDefaultLanguageDetector.php [hunk 8]
```diff
@@ -10,25 +10,18 @@
  */
 class ilDefaultLanguageDetector implements ilLanguageDetector
 {
-    /**
-     * @var ilIniFile
-     */
     protected $ini;
```
Comment (katringross) [inline Services/Language/classes/class.ilDefaultLanguageDetector.php:14]: public function __construct(protected ilIniFile $ini)
    {
    }

File: Services/Language/classes/class.ilHttpRequestsLanguageDetector.php [hunk 9]
```diff
@@ -10,38 +10,30 @@
...
+    protected string $header_value;
 
-    /**
-     * @param array $header_value
-     */
-    public function __construct($header_value)
+    public function __construct(string $header_value)
     {
         $this->header_value = $header_value;
     }
```
Comment (katringross) [inline Services/Language/classes/class.ilHttpRequestsLanguageDetector.php:19]: public function __construct(protected string $header_value)
    {
    }

File: Services/Language/classes/class.ilLangDeprecated.php [hunk 10]
```diff
@@ -42,10 +38,8 @@ public function getDeprecatedLangVars()
      * Get candidates from the db. Base are all lang vars of the
      * english lang file reduced by the ones being accessed (having entries in lng_log)
      */
-    protected function getCandidates()
+    protected function getCandidates(): void
     {
-        $this->candidates = array();
```
Comment (katringross) [inline Services/Language/classes/class.ilLangDeprecated.php:47]: müsste die zeile nicht bleiben? Falls in candidates was drin ist, wird es doch hier geleert oder?

File: Services/Language/classes/class.ilLanguage.php [hunk 11]
```diff
@@ -25,143 +25,41 @@
...
-     * language that is in use
-     * by current user
-     * this language must not be deleted
-     *
-     * @var		string
-     * @access	private
-     */
+    public array $text;
+    public string $lang_default;
     public $lang_user;
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguage.php]: public string $lang_user;

File: Services/Language/classes/class.ilLanguage.php [hunk 12]
```diff
@@ -25,143 +25,41 @@
...
-
-    /**
-     * language full name in that language current in use
-     *
-     * @var		string
-     * @access	private
-     */
+    public string $lang_path;
+    public string $lang_key;
     public $lang_name;
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguage.php]: public string $lang_name;

File: Services/Language/classes/class.ilLanguage.php [hunk 13]
```diff
@@ -433,61 +337,67 @@ public static function lookupId($a_lang_key)
...
     {
         return $this->lang_user;
     }
 
     /**
      * Builds a global default language instance
-     * @return \ilLanguage
      */
     public static function getFallbackInstance()
     {
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguage.php:370]: : self

File: Services/Language/classes/class.ilLanguageDetectorFactory.php [hunk 14]
```diff
@@ -11,24 +11,10 @@ class ilLanguageDetectorFactory
     const DEFAULT_DETECTOR = 1;
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageDetectorFactory.php:11]: missing <?php declare(strict_types=1);

File: Services/Language/classes/class.ilLanguageDetectorFactory.php [hunk 15]
```diff
@@ -11,24 +11,10 @@ class ilLanguageDetectorFactory
...
     const HTTP_REQUEST_DETECTOR = 2;
 
-    /**
-     * @var ilIniFile
-     */
     protected $client_ini;
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageDetectorFactory.php:14]: protected ilIniFile $client_ini;

File: Services/Language/classes/class.ilLanguageDetectorFactory.php [hunk 16]
```diff
@@ -38,17 +24,14 @@ public function __construct()
         $this->settings = $DIC->settings();
     }
 
-    /**
-     * @return ilLanguageDetector[]
-     */
     public function getValidInstances()
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageDetectorFactory.php:30]: : array

File: Services/Language/classes/class.ilLanguageDetectorFactory.php [hunk 17]
```diff
@@ -58,23 +41,21 @@ public function getValidInstances()
     }
 
     /**
-     * @param int $type
      * @throws ilLanguageException
-     * @return ilLanguageDetector
      */
-    public function createDetectorByType($type)
+    public function createDetectorByType(int $type)
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageDetectorFactory.php:48]: : void

File: Services/Language/classes/class.ilLanguageDetectorFactory.php [hunk 18]
```diff
@@ -11,24 +11,10 @@ class ilLanguageDetectorFactory
...
-    /**
-     * @var array
-     */
-    protected $request_information = array();
-
-    /**
-     * @var $ilSettings
-     */
+    protected array $request_information;
     protected $settings;
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageDetectorFactory.php:17]: not sure if it is type ilSettings ???

Comment (katringross) [inline Services/Language/classes/class.ilLanguageDetectorFactory.php:17]: not sure if it is type ilSettings ???

File: Services/Language/classes/class.ilLanguageExtTableGUI.php [hunk 19]
```diff
@@ -5,17 +5,9 @@
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageExtTableGUI.php:5]: missing <?php declare(strict_types=1);

File: Services/Language/classes/class.ilLanguageExtTableGUI.php [hunk 20]
```diff
@@ -5,17 +5,9 @@
...
-    /**
-     * @var array   call parameters
-     */
-    private $params = array();
+    private int $inputsize = 40;
+    private int $commentsize = 30;
+    private array $params;
 
     public function __construct($a_parent_obj, $a_parent_cmd, $a_params = array())
     {
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageExtTableGUI.php:13]: public function __construct(object $a_parent_obj, string $a_parent_cmd, array $a_params = array())

File: Services/Language/classes/class.ilLanguageExtTableGUI.php [hunk 21]
```diff
@@ -61,20 +53,20 @@ protected function fillRow($data)
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageExtTableGUI.php:53]: protected function fillRow(array $data) : void

File: Services/Language/classes/class.ilLanguageExtTableGUI.php [hunk 22]
```diff
@@ -103,7 +95,7 @@ public function initFilter()
         // most filters are only
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageExtTableGUI.php:95]: public function initFilter() : void

File: Services/Language/classes/class.ilLanguageFile.php [hunk 23]
```diff
@@ -34,87 +34,26 @@
...
-
-    /**
-     * header of the language file including the starting line
-     * @var		string
-     */
+    private string $lang_file;
+    private string $lang_key;
+    private string $scope;
     private $header;
-
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageFile.php:67]: private string $header;

File: Services/Language/classes/class.ilLanguageTableGUI.php [hunk 24]
```diff
@@ -28,7 +29,7 @@ public function __construct($a_parent_obj, $a_parent_cmd, $a_folder)
         parent::__construct($a_parent_obj, $a_parent_cmd);
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageTableGUI.php:29]: public function __construct(object $a_parent_obj, string $a_parent_cmd, ilObjLanguageFolder $a_folder)

und dann 

 parent::__construct(object  $a_parent_obj, string $a_parent_cmd);

File: Services/Language/classes/class.ilLanguageTableGUI.php [hunk 25]
```diff
@@ -15,6 +15,7 @@ class ilLanguageTableGUI extends ilTable2GUI
 {
     /** @var ilObjLanguageFolder */
     protected $folder;
+
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageTableGUI.php:18]: protected ilObjLanguageFolder $folder;

File: Services/Language/classes/class.ilLanguageTableGUI.php [hunk 26]
```diff
@@ -69,8 +70,8 @@ public function getItems()
         }
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageTableGUI.php:70]: public function getItems() : void

File: Services/Language/classes/class.ilLanguageTableGUI.php [hunk 27]
```diff
@@ -104,9 +105,9 @@ protected function fillRow($a_set)
...
```
Comment (katringross) [inline Services/Language/classes/class.ilLanguageTableGUI.php:105]: protected function fillRow($a_set) : void

File: Services/Language/classes/class.ilObjLanguage.php [hunk 28]
```diff
@@ -25,7 +25,6 @@ class ilObjLanguage extends ilObject
     public $lang_default;
     public $lang_user;
     public $lang_path;
-
     public $key;
     public $status;
```
Comment (katringross) [inline Services/Language/classes/class.ilObjLanguage.php]: all of type string

File: Services/Language/classes/class.ilObjLanguage.php [hunk 29]
```diff
@@ -129,7 +128,7 @@ public function isUserLanguage()
      */
```
Comment (katringross) [inline Services/Language/classes/class.ilObjLanguage.php:122]: public function __construct(int $a_id = 0, bool $a_call_by_reference = false)

all reteurn types for functions missing

File: Services/Language/classes/class.ilObjLanguage.php [hunk 30]
```diff
@@ -129,7 +128,7 @@ public function isUserLanguage()
...
     public function isInstalled()
     {
```
Comment (katringross) [inline Services/Language/classes/class.ilObjLanguage.php:124]: : bool

File: Services/Language/classes/class.ilObjLanguageAccess.php [hunk 31]
```diff
@@ -148,45 +143,45 @@ public static function _getTranslationLink()
      */
     public static function _isPageTranslation()
     {
```
Comment (katringross) [inline Services/Language/classes/class.ilObjLanguageAccess.php:143]: : bool

File: Services/Language/classes/class.ilObjLanguageExt.php [hunk 32]
```diff
@@ -111,7 +109,7 @@ public function getAllRemarks()
     * @param    array       list of topics
```
Comment (katringross) [inline Services/Language/classes/class.ilObjLanguageExt.php]: public function getCustLangPath() : string

public function getAllRemarks() : array

File: Services/Language/classes/class.ilObjLanguageExt.php [hunk 33]
```diff
@@ -111,7 +109,7 @@ public function getAllRemarks()
...
     * @return   array       module.separator.topic => value
     */
-    public function getAllValues($a_modules = array(), $a_pattern = '', $a_topics = array())
+    public function getAllValues($a_modules = array(), $a_pattern = "", $a_topics = array())
     {
```
Comment (katringross) [inline Services/Language/classes/class.ilObjLanguageExt.php:113]: public function getAllValues(array $a_modules = array(), string $a_pattern = "", array $a_topics = array()) : array

File: Services/Language/classes/class.ilObjLanguageExt.php [hunk 34]
```diff
@@ -126,9 +124,9 @@ public function getAllValues($a_modules = array(), $a_pattern = '', $a_topics =
     * @param    array       list of topics
     * @return   array       module.separator.topic => value
     */
-    public function getChangedValues($a_modules = array(), $a_pattern = '', $a_topics = array())
+    public function getChangedValues($a_modules = array(), $a_pattern = "", $a_topics = array())
     {
```
Comment (katringross) [inline Services/Language/classes/class.ilObjLanguageExt.php:128]: types missing for all comming functions

File: Services/Language/interfaces/interface.ilLanguageDetector.php [hunk 35]
```diff
@@ -10,7 +10,6 @@ interface ilLanguageDetector
 {
```
Comment (katringross) [inline Services/Language/interfaces/interface.ilLanguageDetector.php:10]: missing <?php declare(strict_types=1);

File: Services/Language/classes/Setup/class.ilLanguageMetricsCollectedObjective.php (hunk 2 above)
Comment (iomidev) [reply Services/Language/classes/Setup/class.ilLanguageMetricsCollectedObjective.php]: das habe ich ausgefüllt

File: Services/Language/classes/class.ilLanguageFile.php (hunk 1 above)
Comment (iomidev) [reply Services/Language/classes/class.ilLanguageFile.php]: Das habe ich ausgefüllt

File: Services/Language/interfaces/interface.ilLanguageDetector.php (hunk 35 above)
Comment (katringross) [reply Services/Language/interfaces/interface.ilLanguageDetector.php:10]: @iomidev missing <?php declare(strict_types=1);

File: Services/Language/classes/Setup/class.ilLanguagesInstalledAndUpdatedObjective.php (hunk 3 above)
Comment (iomidev) [reply Services/Language/classes/Setup/class.ilLanguagesInstalledAndUpdatedObjective.php]: it was filled!

File: Services/Language/interfaces/interface.ilLanguageDetector.php (hunk 35 above)
Comment (iomidev) [reply Services/Language/interfaces/interface.ilLanguageDetector.php:10]: It was filled!

File: Services/Language/classes/Setup/class.ilSetupLanguage.php (hunk 5 above)
Comment (iomidev) [reply Services/Language/classes/Setup/class.ilSetupLanguage.php:328]: The type in comment was false.

File: Services/Language/classes/Setup/class.ilSetupLanguage.php (hunk 6 above)
Comment (iomidev) [reply Services/Language/classes/Setup/class.ilSetupLanguage.php:360]: It was changed.

File: Services/Language/classes/class.ilCachedLanguage.php (hunk 7 above)
Comment (iomidev) [reply Services/Language/classes/class.ilCachedLanguage.php:13]: It was be filled

File: Services/Language/classes/class.ilObjLanguageExt.php (hunk 32 above)
Comment (iomidev) [reply Services/Language/classes/class.ilObjLanguageExt.php]: They were be filled

File: Services/Language/classes/class.ilObjLanguageExt.php (hunk 33 above)
Comment (iomidev) [reply Services/Language/classes/class.ilObjLanguageExt.php:113]: It was filled

File: Services/Language/classes/class.ilObjLanguageExt.php (hunk 34 above)
Comment (iomidev) [reply Services/Language/classes/class.ilObjLanguageExt.php:128]: I didn't work in this class yet

**Comment by iomidev** [comment]
PHP 8 Language Service Refactoring

**Pull request merged**